components/bin/build

```diff
@@ -0,0 +1,318 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Builds the lib directory for a component
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Read the configuration for the component
+ */
+const config = JSON.parse(fs.readFileSync(process.argv[2] || 'build.json'));
+
+/**
+ * Extract the configuration values
+ */
+const targets = config.targets || [];                                       // the files to include in the component
+const exclude = new Map((config.exclude || []).map(name => [name, true]));  // files to exclude from the component
+const MATHJAX3 = config.mathjax3 || '../../../mathjax3';                    // path to the mathjax3 .js files
+const LIB = config.lib || './lib';                                          // path to the lib directory to create
+const COMPONENT = config.component || 'part';                               // name of the component
+const GLOBAL = config.global || `../${MATHJAX3}/components/global.js`;      // the location of global.js
+
+/**
+ * The list of files that need to be added to the lib directory
+ */
+let PACKAGE = [];
+
+/**
+ * Process an array of files/directories to be included in the component
+ *
+ * @param {string} base    The root directory for the .ts files
+ * @param {string} dir     The relative path within the root for the files to process
+ * @param {string[]} list  An array of file/directory names to process
+ */
+function processList(base, dir, list) {
+    for (const item of list) {
+        const file = path.join(dir, item);
+        if (!exclude.has(file)) {
+            const stat = fs.statSync(path.resolve(base, file));
+            if (stat.isDirectory()) {
+                processDir(base, file);
+            } else if (file.match(/\.ts/)) {
+                processFile(base, file);
+            }
+        }
+    }
+}
+
+/**
+ * Process the contents of the directory
+ *
+ * @param {string} base    The root directory for the .ts files
+ * @param {string} dir     The relative path within the root for the directory to process
+ */
+function processDir(base, dir) {
+    const root = path.resolve(base, dir);
+    processList(base, dir, fs.readdirSync(root));
+}
+
+/**
+ * Process the contents of a .ts file
+ *
+ * Look for export commands within the file, and determine the objects
+ *   that they reference, then produce the library file that defines
+ *   them, and add the file to the package list.
+ *
+ * @param {string} base    The root directory for the .ts files
+ * @param {string} name    The path of the file relative to the base
+ */
+function processFile(base, name) {
+    console.log(name);
```

Comment:
Does that serve a purpose? If maybe have a meaningful string like `Processing file $(name)`.